src/state/slices/stakingDataSlice/stakingDataSlice.test.ts

```diff
@@ -0,0 +1,522 @@
+
+          const selected = selectSingleValidator(
+            store.getState(),
+            cosmosAccountSpecifier,
+            'cosmosvaloper1xxjvzwjpsf6ktuffkcpx29ut9qfrn0my8xdtqd'
+          )
+          expect(selected).toBeNull()
+        })
+
+        it('returns validator info when validator address is known', async () => {
+          store.dispatch(
+            stakingData.actions.upsertStakingData({
+              accountSpecifier: cosmosAccountSpecifier,
+              stakingData: mockStakingData
+            })
+          )
+
+          const selected = selectSingleValidator(
+            store.getState(),
+            cosmosAccountSpecifier,
+            SHAPESHIFT_VALIDATOR_ADDRESS
+          )
+          expect(selected).toEqual({
+            address: 'cosmosvaloper199mlc7fr6ll5t54w7tts7f4s0cvnqgc59nmuxf',
+            apr: '0.1496681491',
+            commission: '0.100000000000000000',
+            moniker: 'ShapeShift DAO'
+          })
+        })
+      })
+    })
+
+    describe('selectRewardsAmountByAssetId', () => {
+      it('returns empty string when state is empty', async () => {
+        const selected = selectRewardsAmountByAssetId(
+          store.getState(),
+          cosmosAccountSpecifier,
+          SHAPESHIFT_VALIDATOR_ADDRESS,
+          cosmosAssetId
+        )
+        expect(selected).toEqual('')
+      })
+
+      describe('validators and staking data are loaded', () => {
+        beforeEach(() => {
+          store.dispatch(
+            stakingData.actions.upsertValidatorData({
+              validators: mockValidatorData
+            })
+          )
+          store.dispatch(
+            stakingData.actions.upsertStakingData({
+              accountSpecifier: cosmosAccountSpecifier,
+              stakingData: mockStakingData
+            })
+          )
+        })
+
+        it('returns empty string when there is no reward', async () => {
+          const selected = selectRewardsAmountByAssetId(
+            store.getState(),
+            cosmosAccountSpecifier,
+            'cosmosvaloper1qtxec3ggeuwnca9mmngw7vf6ctw54cppey02fs',
+            cosmosAssetId
+          )
+          expect(selected).toEqual('')
+
+          const selected2 = selectRewardsAmountByAssetId(
+            store.getState(),
+            cosmosAccountSpecifier,
+            'cosmosvaloper1r9lxkpqre6j4487ut882xchgr7rdtx3x76gtdp',
+            cosmosAssetId
+          )
+          expect(selected2).toEqual('')
+        })
+
+        it('returns the reward amount when there is a reward', async () => {
+          const selected = selectRewardsAmountByAssetId(
+            store.getState(),
+            cosmosAccountSpecifier,
+            SHAPESHIFT_VALIDATOR_ADDRESS,
+            cosmosAssetId
+          )
+          expect(selected).toEqual('3.831752143667562385')
+        })
+      })
+    })
+
+    describe('selectTotalBondingsBalanceByAssetId', () => {
+      it('returns 0 when state is empty', async () => {
```

Comment:
Making it clear we return numbers as strings
```suggestion
      it('returns '0' when state is empty', async () => {
```